src/configure/helper/commonHelper.ts

```diff
@@ -11,6 +16,32 @@ export async function sleepForMilliSeconds(timeInMs: number): Promise<void> {
     });
 }
 
+export async function createOrUpdateGithubSecret(secretName: string, body: string): Promise < void> {
+    let secretKeyObject: GitHubSecretKey = await this._getGitHubSecretKey();
+    let sodiumObj = new SodiumLibHelper(secretKeyObject.key);
+    let encryptedBytes: Uint8Array = sodiumObj.encrypt(body);
+    let encryptedBytesAsString: string = SodiumLibHelper.convertUint8ArrayToString(encryptedBytes);
+    let encryptedEncodedText = SodiumLibHelper.encodeToBase64(encryptedBytesAsString);
+    await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
+}
+
+export async function generateGitHubRepository(orgName: string, localPath: string, githubClient: GithubClient): Promise<GitHubRepo | void>{
+    let repoName = localPath.substring(localPath.lastIndexOf("\\") + 1);
+    let repoDetails = await githubClient.createGithubRepo(orgName, repoName) as GitHubRepo | void;
+    // Case : GitHub Repository name is same as the local repo
+    if(repoDetails){
+        return repoDetails; 
+    }
+    //Case: Local repository name is not available, therefore organization name has been appended
+    repoName = repoName + "_" + orgName;
+    repoDetails = await githubClient.createGithubRepo(orgName, repoName) as GitHubRepo | void;
+    if(repoDetails){
+        return repoDetails; 
+    }
+    //Case: If the above two repository names are not available, uuid is also appended
+    return await githubClient.createGithubRepo(orgName, repoName + "-" + uuid().substr(0,5));  
```

Comment:
it would be weird to stop the flow without telling user how they can fix it. Either we ask the repo name from user or append a unique guid. Unique guid seems to be simpler approach. And the name will still be helpful as it starts with org+repo name